nipype/pipeline/engine/tests/test_newnode_neuro.py

```diff
@@ -0,0 +1,133 @@
+
+#dj niworkflows vs ...??
+from nipype.interfaces.utility import Rename
+import nipype.interfaces.freesurfer as fs
+
+from fmriprep.interfaces.freesurfer import PatchedConcatenateLTA as ConcatenateLTA
+
+import pdb
+
+Name = "example"
+DEFAULT_MEMORY_MIN_GB = None
+# TODO, adding fields to Inputs (subject_id)
+Inputs = {"subject_id": "sub-01",
+          "output_spaces": ["fsaverage", "fsaverage5"],
+          "source_file": "/Users/dorota/fmriprep_test/workdir1/fmriprep_wf/single_subject_01_wf/func_preproc_ses_test_task_fingerfootlips_wf/bold_t1_trans_wf/merge/vol0000_xform-00000_merged.nii",
+          "t1_preproc": "/Users/dorota/fmriprep_test/output1/fmriprep/sub-01/anat/sub-01_T1w_preproc.nii.gz",
+          "t1_2_fsnative_forward_transform": "/Users/dorota/fmriprep_test/workdir1/fmriprep_wf/single_subject_01_wf/anat_preproc_wf/surface_recon_wf/t1_2_fsnative_xfm/out.lta",
+          "subjects_dir": "/Users/dorota/fmriprep_test/fmriprep_test/output1/freesurfer/"
+}
+
+def test_neuro():
+
+    # wf = Workflow(name, mem_gb_node=DEFAULT_MEMORY_MIN_GB,
+    #               inputs=['source_file', 't1_preproc', 'subject_id',
+    #                       'subjects_dir', 't1_2_fsnative_forward_transform',
+    #                       'mem_gb', 'output_spaces', 'medial_surface_nan'],
+    #               outputs='surfaces')
+    #
+    #dj: why do I need outputs?
+
+    pdb.set_trace()
+    wf = NewWorkflow(name=Name, inputs=Inputs, workingdir="test_neuro")
+    pdb.set_trace()
+
+
+    # @interface
+    # def select_target(subject_id, space):
+    #     """ Given a source subject ID and a target space, get the target subject ID """
+    #     return subject_id if space == 'fsnative' else space
+
+    # TODO: shouldn't map with subject?
+    def select_target(subject_id, space):
+        """ Given a source subject ID and a target space, get the target subject ID """
+        return subject_id if space == 'fsnative' else space
+
+    #select_target_interface = Function_Interface(select_target, ["out"])
+
+
+    # wf.add('targets', select_target(subject_id=wf.inputs.subject_id))
+    #   .map('space', space=[space for space in wf.inputs.output_spaces
+    #                        if space.startswith('fs')])
+
+    #dj: don't have option in map to connect with wf input
+
+    wf.add(runnable=select_target, name="targets", subject_id="subject_id")\
+        .map(mapper="space", inputs={"space": [space for space in Inputs["output_spaces"]
+                                               if space.startswith("fs")]})
+
+
+    # wf.add('rename_src', Rename(format_string='%(subject)s',
+    #                             keep_ext=True,
+    #                             in_file=wf.inputs.source_file))
+    #   .map('subject')
+
+    pdb.set_trace()
+    wf.add(name='rename_src',
+           runnable=Rename(format_string='%(subject)s', keep_ext=True),
+                                in_file="source_file", subject="subject_id")\
+        .map('subject')
```

Comment:
I think here what we actually want is:

```Python
wf.add('rename_src', Rename(format_string='%(subject)s', keep_ext=True),
       in_file="inputs.source_file", subject="select_target.out")
```

The mapping is already being done at `select_target`. Since these are no longer `MapNode`s, we don't need to re-map unless we join first.